lms/models/assignment.py

```diff
@@ -24,23 +24,11 @@ class Assignment(BASE):
     __tablename__ = "assignment"
     __table_args__ = (
         sa.UniqueConstraint("resource_link_id", "tool_consumer_instance_guid"),
-        sa.UniqueConstraint("tool_consumer_instance_guid", "ext_lti_assignment_id"),
-        sa.CheckConstraint(
-            "NOT(resource_link_id IS NULL AND ext_lti_assignment_id IS NULL)",
-            name="nullable_resource_link_id_ext_lti_assignment_id",
-        ),
     )
 
     id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
 
-    ext_lti_assignment_id = sa.Column(sa.UnicodeText, nullable=True)
-    """
-    ID given for the assignment by Canvas.
-
-    Null for non canvas assignments and older Canvas ones.
-    """
-
-    resource_link_id = sa.Column(sa.Unicode, nullable=True)
+    resource_link_id = sa.Column(sa.Unicode, nullable=False)
```

Comment:
I think you mentioned we would have to delete some rows to make this true? What happens if you change a constraint that invalidates existing rows anyway?